src/Microsoft.Health.Dicom.Core/Features/Store/ValidationResultBuilder.cs

```diff
@@ -0,0 +1,104 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Store;
+
+internal sealed class ValidationResultBuilder : IValidationResultBuilder
+{
+    private readonly IList<string> _errorMessages;
```

Comment:
nit: do not use interfaces for private members that do not leverage any polymorphism; no need to perform virtual dispatches